platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,528 @@
+then followed by a software-only upgrade to adopt the `config.txt` file using that state. This will change to the
+use of a Genesis Roster or Override Roster. The new startup sequence will be determined by the presence of a Genesis
+Roster, State, or Override Roster as described in the following sections.
+
+### The new Startup Behaviour
+
+![](TSS%20Roster%20Lifecycle-Proposed%20Startup%20Behavior.drawio.svg)
+
+#### Genesis Roster
+
+A `Genesis Roster` is an optional Roster that DevOps may provide for the explicit purpose of starting a Genesis Network
+Process. It is essential to distinguish the Genesis Roster from the Candidate, Active, or Override rosters. The
+Genesis Roster is a special roster used for the sole purpose of bootstrapping a brand-new Genesis network; one in
+which there is no existing State, and the round number begins at 1. However, its structure is exactly the same as
+the Active or Candidate Rosters.
+
+The equivalent of the Genesis Roster in the current DevOps flow is the `config.txt` file. DevOps and Services may
+choose to create the Genesis Roster from the `config.txt` file or some other mechanism. The trigger for this
+workflow will be the presence of a Genesis Roster at start-up with no State on disk.
+
+A new method will be added to the `PlatformBuilder` that will be used by the App to set the Genesis Roster.
+In most cases, there will be no need to call the method on `PlatformBuilder` to set the Genesis Roster except in the
+events we need to bootstrap a Genesis Network.
+
+```java
+//in PlatformBuilder
+
+/*
+  Set the Genesis Roster for the network.
+  <p>
+  This method is used to set the optional Genesis Roster for the network.
+  The Genesis Roster is a special roster used for the sole-purpose of bootstrapping a network.
+  The Genesis Roster, if set, is immutable and will be used to start a Genesis network Process or Network Transplant process.
+  <p>
+  @param genesisRoster the Genesis Roster for the network.
+ */
+PlatformBuilder withGenesisRoster(@NonNull final Roster genesisRoster);
+```
+
+The logic to determine whether to invoke this method could be determined by some mechanism Services deems appropriate.
+
+#### Override Roster
+
+An `Override Roster` is an optional Roster that DevOps may provide for the explicit purpose of starting a network using
+a specific state, with a specific set of nodes, such as during a Network Transplant Process. A common example of
+this is Mainnet State being transplanted into a testing network. Provided an existing State and an `Override Roster`,
+the platform will adopt that roster.
+
+The next round number for this network will continue from the last round number in the provided state.
+The trigger for adopting the `Override Roster` will be the presence of an Override Roster at start-up with a State
+existing on disk. Bear in mind the Network Transplant process is designed primarily with test networks in mind.
```

Comment:
An upgrade is still required to adopt the override roster because, until we have birth rounds enabled, the software version is the only mechanism by which we determine which roster to use to verify the signatures of events in the PCES written by nodes in the roster being replaced. The consensus node must be able to determine which roster (the override roster or the one originally in the state on disk) to use to verify the signatures of these events. Without a birth round or a software change to make that decision, the verification of PCES will fail.

I do not want to hold this proposal up any further, so please create a ticket to update this proposal after it has been merged and I will resolve this comment and approve.

Reply:
Given an existing network A, and a test network B that must be transplanted, with A and B running the same software version. Seems like a valid use-case, especially in testing. Will this imply we don't support transplanting A into B _without_ a software version bump, at least until Birth Rounds or DAB (whichever comes first?)?

Reply:
Correct, today the process requires a config-only version upgrade when transplanting into a test network, specifically because of the problem I described.

Reply:
updated the diagram and text to reflect this.